presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/StaticHiveCluster.java

```diff
@@ -67,17 +73,22 @@ public HiveMetastoreClient createMetastoreClient()
         List<HostAndPort> metastores = new ArrayList<>(addresses);
         Collections.shuffle(metastores.subList(1, metastores.size()));
 
-        TTransportException lastException = null;
+        TException lastException = null;
         for (HostAndPort metastore : metastores) {
             try {
-                return clientFactory.create(metastore);
+                HiveMetastoreClient hiveMetastoreClient = clientFactory.create(metastore);
+                if (!com.google.common.base.Strings.isNullOrEmpty(hiveUserName)) {
+                    hiveMetastoreClient.setUGI(hiveUserName, new ArrayList<String>());
+                }
+                return hiveMetastoreClient;
             }
-            catch (TTransportException e) {
+            catch (TException e) {
                 lastException = e;
             }
         }
 
-        throw new PrestoException(HIVE_METASTORE_ERROR, "Failed connecting to Hive metastore: " + addresses, lastException);
+        throw new PrestoException(HIVE_METASTORE_ERROR, format("Failed connecting to Hive metastore: %s use %s ", addresses, hiveUserName),
```

Comment:
what if hiveUserName is null

Reply:
do you suggest like this?
![image](https://user-images.githubusercontent.com/9247964/35214157-2adfdff0-ff9b-11e7-9ea9-f855411cc737.png)